gaeguli/pipeline.c

```diff
@@ -277,14 +280,48 @@ gaeguli_pipeline_new (void)
 
   pipeline =
       gaeguli_pipeline_new_full (DEFAULT_VIDEO_SOURCE,
-      DEFAULT_VIDEO_SOURCE_DEVICE);
+      DEFAULT_VIDEO_SOURCE_DEVICE, DEFAULT_ENCODING_METHOD);
 
   return g_steal_pointer (&pipeline);
 }
 
+struct encoding_method_params
+{
+  const gchar *pipeline_str;
+  GaeguliEncodingMethod encoding_method;
+  GaeguliVideoCodec codec;
+};
+
+static struct encoding_method_params enc_params[] = {
+  {GAEGULI_PIPELINE_GENERAL_H264ENC_STR, GAEGULI_ENCODING_METHOD_GENERAL,
+      GAEGULI_VIDEO_CODEC_H264},
+  {GAEGULI_PIPELINE_GENERAL_H265ENC_STR, GAEGULI_ENCODING_METHOD_GENERAL,
+      GAEGULI_VIDEO_CODEC_H265},
+  {GAEGULI_PIPELINE_NVIDIA_TX1_H264ENC_STR, GAEGULI_ENCODING_METHOD_NVIDIA_TX1,
+      GAEGULI_VIDEO_CODEC_H264},
+  {GAEGULI_PIPELINE_NVIDIA_TX1_H265ENC_STR, GAEGULI_ENCODING_METHOD_NVIDIA_TX1,
+      GAEGULI_VIDEO_CODEC_H265},
+  {NULL, 0, 0},
+};
+
+static const gchar *
+_lookup_enc_string (GaeguliEncodingMethod encoding_method,
+    GaeguliVideoCodec codec)
+{
+  struct encoding_method_params *params = enc_params;
+
+  for (; params->pipeline_str != NULL; params++) {
+    if (params->encoding_method == encoding_method && params->codec == codec)
+      return params->pipeline_str;
+  }
+
+  return NULL;
+}
+
 static GstElement *
-_build_target_pipeline (GaeguliVideoCodec codec, const gchar * fifo_path,
-    GError ** error)
+_build_target_pipeline (GaeguliEncodingMethod encoding_method,
+    GaeguliVideoCodec codec, GaeguliVideoResolution resolution,
```

Comment:
You're not using the passed resolution anywhere inside the function. I also thought it's expected all target pipelines use the resolution of video source, which makes this parameter even more useless.

Reply:
Indeed, I forgot to remove resolution here. 
I was thinking of different resolution per target, but I found it's unmanageable if I deploy two encoders on nvidia tx1 board. I'll put it on hold until I find a solution.